apps/api/src/domain/domainModelValidators/contextValidators/textFieldContext.validator.ts

```diff
@@ -0,0 +1,43 @@
+import { InternalError } from 'apps/api/src/lib/errors/InternalError';
+import { TextFieldContext } from '../../models/context/text-field-context/text-field-context.entity';
+import { EdgeConnectionContextType } from '../../models/context/types/EdgeConnectionContextType';
+import { isNullOrUndefined } from '../../utilities/validation/is-null-or-undefined';
+import EmptyTextTargetFieldInContextError from '../errors/context/EmptyTextTargetFieldInContextError';
+import InvalidCharRangeError from '../errors/context/InvalidCharRangeError';
+import InvalidEdgeConnectionContextError from '../errors/context/InvalidEdgeConnectionContextError';
+import NullOrUndefinedEdgeConnectionContextDTOError from '../errors/context/NullOrUndefinedEdgeConnectionContextDTOError';
+import { Valid } from '../Valid';
+
+type Integer = number;
+
+/**
+ * TODO [https://www.pivotaltracker.com/story/show/181891200]
+ * We should create our own validation library.
+ */
+const isInteger = (input: unknown): input is Integer => Number.isInteger(input);
+
+export const textFieldContextValidator = (input: unknown): Valid | InternalError => {
+    if (isNullOrUndefined(input))
+        return new NullOrUndefinedEdgeConnectionContextDTOError(
+            EdgeConnectionContextType.textField
+        );
+
+    const allErrors: InternalError[] = [];
+
+    const { target, charRange } = input as TextFieldContext;
```

Comment:
This casting is really lame!

Reply:
I'll bet in this case it's actually necessary....

Reply:
Bingo. It's casting to broaden `unknown` for the purpose of validation. This is one of the acceptable uses of casting. But thanks for calling me on my stuff!